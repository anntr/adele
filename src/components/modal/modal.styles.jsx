import styled from 'styled-components';
import { modalControls, modal } from '../../style_tokens/tokens';

const fixedPositioning = `
position: fixed;
top: 0;
bottom: 0;
left: 0;
right: 0;
z-index: 100000
`;

const StyledModal = styled.div`
  font-family: ${modal.typography.fontFamily};
  padding: 30px; 
  background-color: ${modal.background};
  width: 60vw;
  height: 60vh;
  margin: auto;
  ${fixedPositioning}
`;

const StyledModalBackdrop = styled.div`
  position: fixed;
  background-color: ${modal.shadow};
  opacity: 0.5;
  ${fixedPositioning}
`;

const StyledModalCloseButton = styled.button`
  border: none;
  outline: none;
  background: ${modalControls.background};
  width: 32px;
  height: 32px;
  border-radius: 32px;
  cursor: pointer;
  float: right; 
`;

export { StyledModal, StyledModalBackdrop, StyledModalCloseButton };
