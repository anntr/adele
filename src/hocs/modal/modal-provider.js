import { compose, getContext, withContext, withHandlers, withState } from 'recompose';
import PropTypes from 'prop-types';

const modalContextType = {
  openModal: PropTypes.func.isRequired,
};

const withModal = getContext(modalContextType);

const modalProvider = compose(
  withState('component', 'setComponent', undefined),
  withState('isVisible', 'setVisible', false),
  withHandlers({
    openModal: ({ setVisible, setComponent }) => (ContentComponent) => {
      setComponent(() => ContentComponent);
      setVisible(true);
    },
    closeModal: ({ setVisible }) => () => setVisible(false),
  }),
  withContext(modalContextType, ({ openModal }) => ({ openModal })),
);

export { modalProvider, withModal };
