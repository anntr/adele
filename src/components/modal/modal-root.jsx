import React from 'react';
import PropTypes from 'prop-types';
import Modal from './modal';
import { modalProvider } from '../../hocs/modal/modal-provider';

const ModalRoot = modalProvider(({
  component: Component, closeModal, isVisible, children,
}) => (
  <div>
    {children}
    {isVisible &&
    <Modal onClose={closeModal}>
      <Component />
    </Modal>}
  </div>
));

ModalRoot.propTypes = {
  children: PropTypes.node.isRequired,
  component: PropTypes.func,
};

ModalRoot.defaultProps = {
  component: undefined,
};

export { ModalRoot as default };
