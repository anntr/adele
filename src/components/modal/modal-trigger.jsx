import { withModal } from '../../hocs/modal/modal-provider';

const ModalTrigger = withModal(({ children, content, openModal }) =>
  children(() => openModal(content)));

export { ModalTrigger as default };
