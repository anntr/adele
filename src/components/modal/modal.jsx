import React from 'react';
import PropTypes from 'prop-types';

import { StyledModal, StyledModalBackdrop, StyledModalCloseButton } from './modal.styles';
import Icon from '../icon/icon';
import close from '../../icons/close.svg';
import { modalControls } from '../../style_tokens/tokens';

const Modal = ({ children, onClose }) => (
  <div>
    <StyledModalBackdrop onClick={onClose} />
    <StyledModal>
      <StyledModalCloseButton onClick={onClose}>
        <Icon i={close} size="xs" color={modalControls.color} />
      </StyledModalCloseButton>
      {children}
    </StyledModal>
  </div>
);

Modal.propTypes = {
  children: PropTypes.node.isRequired,
  onClose: PropTypes.func.isRequired,
};

export { Modal as default };
