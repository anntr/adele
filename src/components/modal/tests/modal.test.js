import React from 'react';
import { StyledModalCloseButton, StyledModalBackdrop } from '../modal.styles';
import Modal from '../modal';

const MockChild = () => <div />;
const mockFunc = jest.fn();

afterEach(() => mockFunc.mockClear());

it('renders modal correctly', () => {
  const wrapper = mount(<Modal onClose={mockFunc}><MockChild /></Modal>);

  expect(wrapper).toMatchSnapshot();
});

it('close button calls callback', () => {
  const wrapper = shallow(<Modal onClose={mockFunc}><MockChild /></Modal>);

  wrapper.find(StyledModalCloseButton).simulate('click');

  expect(mockFunc).toBeCalled();
});

it('can be closed by clicking outside of modal', () => {
  const wrapper = shallow(<Modal onClose={mockFunc}><MockChild /></Modal>);

  wrapper.find(StyledModalBackdrop).simulate('click');

  expect(mockFunc).toBeCalled();
});
