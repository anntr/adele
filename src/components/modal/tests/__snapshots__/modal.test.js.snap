// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders modal correctly 1`] = `
<Modal
  onClose={[Function]}
>
  <div>
    <styled.div
      onClick={[Function]}
    >
      <div
        className="sc-bwzfXH bPTWRz"
        onClick={[Function]}
      />
    </styled.div>
    <styled.div>
      <div
        className="sc-bdVaJa jnSnMc"
      >
        <styled.button
          onClick={[Function]}
        >
          <button
            className="sc-htpNat cNCcHG"
            onClick={[Function]}
          >
            <Icon
              active={false}
              color="#777777"
              i="test-file-stub"
              in="blue"
              rotate={0}
              size="xs"
            >
              <Styled(SVGInline)
                color="#777777"
                data-active={false}
                in="blue"
                rotate={0}
                size="xs"
                svg="test-file-stub"
              >
                <SVGInline
                  className="sc-bxivhb dAGAdf"
                  classSuffix="-svg"
                  cleanup={Array []}
                  cleanupExceptions={Array []}
                  color="#777777"
                  component="span"
                  data-active={false}
                  in="blue"
                  rotate={0}
                  size="xs"
                  svg="test-file-stub"
                >
                  <span
                    className="SVGInline sc-bxivhb dAGAdf"
                    color="#777777"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "test-file-stub",
                      }
                    }
                    data-active={false}
                    in="blue"
                    rotate={0}
                    size="xs"
                  />
                </SVGInline>
              </Styled(SVGInline)>
            </Icon>
          </button>
        </styled.button>
        <MockChild>
          <div />
        </MockChild>
      </div>
    </styled.div>
  </div>
</Modal>
`;
